source/components/JsxParser.test.tsx

```diff
@@ -1238,4 +1238,76 @@ describe('JsxParser Component', () => {
 			expect(component.ParsedChildren[0].props.children[1].key).toBeFalsy()
 		})
 	})
+
+	describe('Functions', () => {
+		it('supports nested jsx inside arrow functions', () => {
+			// see
+			// https://astexplorer.net/#/gist/fc48b12b8410a4ef779e0477a644bb06/cdbfc8b929b31e11e577dceb88e3a1ee9343f68e
+			// for acorn AST
+			const { html } = render(
+				<JsxParser
+					components={{ Custom }}
+					bindings={{ items: [1, 2] }}
+					jsx="{items.map(item => <Custom><p>{item}</p></Custom>)}"
+				/>,
+			)
+			expect(html).toMatch('<div class="jsx-parser"><div><p>1</p></div><div><p>2</p></div></div>')
+		})
+
+		it('supports JSX expressions inside arrow functions', () => {
+			const { html } = render(
+				<JsxParser
+					components={{ Custom }}
+					bindings={{ items: [{ name: 'Megeara', title: 'Fury' }] }}
+					jsx="{items.map(item => <Custom text={item.title}><p>{item.name}</p></Custom>)}"
+				/>,
+			)
+			expect(html).toMatch('<div class="jsx-parser"><div>Fury<p>Megeara</p></div></div>')
+		})
+
+		it('passes attributes', () => {
+			const PropTest = (props: { booleanAttribute: boolean}) => <>{`val:${props.booleanAttribute}`}</>
+			const { html } = render(
+				<JsxParser
+					renderInWrapper={false}
+					components={{ PropTest }}
+					bindings={{ items: [
+						{ name: 'Megeara', friend: true },
+						{ name: 'Austerious', friend: false },
+					] }}
+					jsx="{items.map(item => <p><PropTest booleanAttribute={item.friend} /></p>)}"
+				/>,
+			)
+			expect(html).toEqual('<p>val:true</p><p>val:false</p>')
+		})
+
+		it('passes spread attributes', () => {
+			const PropTest = (props: any) => <>{JSON.stringify(props)}</>
+			const { html } = render(
+				<JsxParser
+					renderInWrapper={false}
+					components={{ PropTest }}
+					bindings={{ items: [
+						{ name: 'Megeara', friend: true },
+					] }}
+					jsx="{items.map(item => <PropTest {...item} />)}"
+				/>,
+			)
+			expect(html).toEqual('{"name":"Megeara","friend":true}')
+		})
+
+		it.skip('[NOT IMPLEMENTED: included for PR discussion] supports function expressions with nested jsx', () => {
+			// this doesn't work because we need to support
+			// ReturnStatement + BlockStatement
```

Comment:
Yeah - I think that starting down this rabbit-hole, as you mentioned, begins the perilous journey into parsing JS instead of JSX, which we should avoid. The inline arrow-function style binding is great, because it allows the `.map()` but this opens a can of worms.

We originally discussed having this style be acceptable, as long as the _only_ statement inside the body of the function was a return statement - at which point it's functionally equivalent to the above test-cases, but IMHO it's simpler to just leave it out and ONLY allow the arrow-function syntax with no body.